client/lib/feed-post-store/actions.js

```diff
@@ -14,7 +14,7 @@ feedPostBatcher = new PostBatcher( {
 	buildUrl: function( postKey ) {
 		return '/read/feed/' + encodeURIComponent( postKey.feedId ) + '/posts/' + encodeURIComponent( postKey.postId );
 	},
-	resultKeyRegex: /\/read\/feed\/(\d+)\/posts\/(\d+)/,
+	resultKeyRegex: /\/read\/feed\/(\w+)\/posts\/(\w+)/,
```

Comment:
While valid feed/blog ID and feed/blog post ID contains only digits, a user can navigate to feed/blog with letters in its ID, which causes 404 to be shown, but we need to modify the regexp here for that to work.